brayns/common/light/Light.h

```diff
@@ -18,40 +17,43 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
-#ifndef LIGHT_H
-#define LIGHT_H
+#pragma once
 
 #include <brayns/api.h>
+#include <brayns/common/PropertyObject.h>
 #include <brayns/common/types.h>
 
+SERIALIZATION_ACCESS(Light)
+
 namespace brayns
 {
 /**
-    Light object
+ * @brief The LightType enum defines the different types of light
+ */
+enum class LightType
+{
+    SPHERE,
+    DIRECTIONAL,
+    QUAD,
+    SPOTLIGHT
+};
 
-    This object is an abstract class defining a light source with a color and an
-    intensity. This is the base class for any light source available in the
-    scene (Point, Directional, etc)
+/**
+ * @brief The Light class defines a light source by type, id and properties
  */
-class Light
+class Light : public PropertyMap
 {
 public:
-    BRAYNS_API virtual ~Light() = 0;
-
-    /** Light source RGB color */
-    BRAYNS_API void setColor(const Vector3f& color) { _color = color; }
-    BRAYNS_API const Vector3f getColor() const { return _color; }
-    /** Light source intensity */
-    BRAYNS_API void setIntensity(const float intensity)
-    {
-        _intensity = intensity;
-    }
-    BRAYNS_API float getIntensity() const { return _intensity; }
-protected:
-    BRAYNS_API Light(const Vector3f& color, float intensity);
+    Light() = default;
+    Light(const LightType type, const Vector3f& color, const float intensity,
+          const int id);
```

Comment:
Without considering how the RPC interface would look for the moment, is it necessary to make the ID part of the light? Shouldn't `LightManager::addLight` return the ID instead?
That seems more consistent with the model API in my opinion. Models don't contain their own ID, it belong to `ModelDescriptor` and  which is returned by `Scene::addModel`.

Reply:
Now considering the json serialization, it seems easier the way you did it, but I'm not super happy about the idea of having the ID in the light. Anyway, I would remove the id from the constructor, make LightManager a friend and the only class capable of setting the ID and make LightManager::addLight return the ID.